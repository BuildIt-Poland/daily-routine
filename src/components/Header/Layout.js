import styled from 'styled-components';

import { colorWhite, spacingMedium } from '../../styles/designTokens';

const Layout = styled.header`
  width: 100%;
  background: ${colorWhite};
  height: 6rem;
  display: flex;
  align-items: center;
  padding: 0 ${spacingMedium};
`;

export default Layout;
