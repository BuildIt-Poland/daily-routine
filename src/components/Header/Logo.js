import React from 'react';
import { Link as LinkBase } from 'react-router-dom';
import styled from 'styled-components';

import { colorBlack, spacingSmall, borderWidthThin, fontSmall } from '../../styles/designTokens';
import { Logo as LogoIcon } from '../Icons';

const Link = styled(LinkBase)`
  display: flex;
  text-decoration: none;
  align-items: center;
  height: 100%;
  cursor: pointer;
`;

const LogoWrapper = styled.div`
  padding-top: 0.6rem;
  padding-right: ${spacingSmall};
  margin-right: ${spacingSmall};
  border-right: ${borderWidthThin} solid ${colorBlack};
`;

const Tagline = styled.p`
  margin: 0;
  padding: 0;
  color: ${colorBlack};
  font-size: 1.2rem;
  line-height: 1;
  text-transform: uppercase;
`;

function Logo() {
  return (
    <Link to="/">
      <LogoWrapper>
        <LogoIcon />
      </LogoWrapper>
      <Tagline>
        Daily scrum <br />
        absurd answers
      </Tagline>
    </Link>
  );
}

export default Logo;
