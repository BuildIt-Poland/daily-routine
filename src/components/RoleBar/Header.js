import styled from 'styled-components';

import media from '../../styles/media';
import { spacingMedium } from '../../styles/designTokens';

const Header = styled.header`
  display: grid;
  grid-template-columns: 1fr auto;
  align-items: center;
  /* compensate for medium spacing and gear icon width with left padding */
  padding: 0 1.4rem 0 5.4rem;
  height: 5rem;
  background-color: rgba(255, 255, 255, 0.5);

  ${media.phone`
    padding-left: ${spacingMedium};
  `};

  ${media.smallPhone`
      height: 4rem;
  `};
`;

export default Header;
