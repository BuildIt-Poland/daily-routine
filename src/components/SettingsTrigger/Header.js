import React from 'react';

import Subheadline from '../Subheadline';

function Header() {
  return (
    <Subheadline>
      My role <strong>settings</strong>
    </Subheadline>
  );
}

export default Header;
