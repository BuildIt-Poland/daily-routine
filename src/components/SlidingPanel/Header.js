import styled from 'styled-components';

import { spacingMedium } from '../../styles/designTokens';

const Header = styled.header`
  height: 6rem;
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 ${spacingMedium};
  background-color: ${({ theme }) => theme.primaryColor};
`;

export default Header;
