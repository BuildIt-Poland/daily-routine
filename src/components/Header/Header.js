import React from 'react';

import Wrapper from './Wrapper';
import Logo from './Logo';

function Header() {
  return (
    <Wrapper role="banner">
      <Logo />
    </Wrapper>
  );
}

export default Header;
