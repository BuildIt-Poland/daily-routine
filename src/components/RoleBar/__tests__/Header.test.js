import React from 'react';
import { create } from 'react-test-renderer';

import Header from '../Header';

describe('COMPONENT - RoleBar Header', () => {
  it('renders correctly', () => {
    const component = create(<Header />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
