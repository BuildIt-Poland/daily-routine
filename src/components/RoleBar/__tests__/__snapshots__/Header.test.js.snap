// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - RoleBar Header renders correctly 1`] = `
.c0 {
  display: grid;
  grid-template-columns: 1fr auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 1.4rem 0 5.4rem;
  height: 5rem;
  background-color: rgba(255,255,255,0.5);
}

@media (max-width:31.25em) {
  .c0 {
    padding-left: 2rem;
  }
}

<header
  className="c0"
/>
`;
